applications/CoSimulationApplication/custom_python/add_co_sim_io_to_python.cpp

```diff
@@ -0,0 +1,372 @@
+// KRATOS  / ___|___/ ___|(_)_ __ ___  _   _| | __ _| |_(_) ___  _ ___
+//        | |   / _ \___ \| | '_ ` _ \| | | | |/ _` | __| |/ _ \| '_  |
+//        | |__| (_) |__) | | | | | | | |_| | | (_| | |_| | (_) | | | |
+//         \____\___/____/|_|_| |_| |_|\__,_|_|\__,_|\__|_|\___/|_| |_|
+//
+//  License:		 BSD License
+//					 license: CoSimulationApplication/license.txt
+//
+//  Main authors:    Philipp Bucher
+//
+
+// System includes
+#include<unordered_map>
+
+// External includes
+
+// Project includes
+#include "includes/model_part.h"
+#include "custom_python/add_co_sim_io_to_python.h"
+
+// CoSimIO
+// defining the macros that are used within the CoSimIO to point the the Kratos-native macros
+// Note that this has to be done BEFORE including the header the first time
+#define KRATOS_CO_SIM_INFO KRATOS_INFO
+#define KRATOS_CO_SIM_INFO_IF KRATOS_INFO_IF
+#define KRATOS_CO_SIM_ERROR KRATOS_ERROR
+#define KRATOS_CO_SIM_ERROR_IF KRATOS_ERROR_IF
+#define KRATOS_CO_SIM_ERROR_IF_NOT KRATOS_ERROR_IF_NOT
+#include "co_simulation_api/co_sim_io.h"
```

Comment:
Here I define the macros before including the header. This way I can use the macros of the including code
Not sure if this is what you want / intended @pooyan-dadvand

Reply:
I was thinking a little bit different but this also does the same!